app_flair.py

```diff
@@ -116,126 +126,185 @@ def parse_prediction_request(json_object: JsonDict) -> PredictionRequest:
+        token.add_tag("ner", "O")
+        i += 1
+        tokens.append(token)
+    return False, tokens
+
+
+def flair_train_ner(train: list, dev: list):
     global tagger_ner
     corpus = Corpus(train=train, dev=dev, test="")
-    trainer: ModelTrainer = ModelTrainer(tagger_ner, corpus)
-    trainer.train('model_ner', learning_rate=0.1, mini_batch_size=8, max_epochs=1,
-                  save_final_model=False)
-    tagger_ner = SequenceTagger.load('model_ner/best-model.pt')
+
+    tagger_ner_for_train = copy.deepcopy(tagger_ner)
+    trainer: ModelTrainer = ModelTrainer(tagger_ner_for_train, corpus)
+    trainer.train(
+        "model_ner",
+        learning_rate=0.1,
+        mini_batch_size=8,
+        max_epochs=1,
+        save_final_model=False,
+    )
+    tagger_ner = SequenceTagger.load("model_ner/best-model.pt")
     lock_ner_train.release()
 
 
-def parse_pos_train_request(json_object: JsonDict):
-    documents = json_object['documents']
-    tagset = [b.decode('utf-8') for b in list(tagger_pos.tag_dictionary.idx2item)]
+def parse_pos_train_request(json_object: JsonDict) -> (list, list):
+    documents = json_object["documents"]
     list_sentences = []
+    typesystem = load_typesystem(json_object["typeSystem"])
     for document in documents:
-        typesystem = load_typesystem(json_object["typeSystem"])
-        cas = load_cas_from_xmi(document['xmi'], typesystem=typesystem)
-        sentence_list = cas.select(SENTENCE_TYPE)
-        for sentence in sentence_list:
-            tokens = []
-            pos_list = list(cas.select_covered(POS_TYPE, sentence))
-            token_list = list(cas.select_covered(TOKEN_TYPE, sentence))
-            if len(pos_list) != len(token_list):
-                continue
-            flag = False
-            for pos in pos_list:
-                if pos.PosValue not in tagset:
-                    flag = True
-                    break
-                token = Token(cas.get_covered_text(pos))
-                token.add_tag('pos', pos.PosValue)
-                tokens.append(token)
-            if flag:
-                continue
-            s = Sentence()
-            s.tokens = tokens
-            list_sentences.append(s)
+        cas = load_cas_from_xmi(document["xmi"], typesystem=typesystem)
+        list_from_document = flair_train_pos_dataset(cas)
+        list_sentences.extend(list_from_document)
     train, dev = train_test_split(list_sentences, train_size=0.8)
     return train, dev
 
 
-def train_pos(train, dev):
+def flair_train_pos_dataset(cas: Cas):
+    tagset = [b.decode("utf-8") for b in list(tagger_pos.tag_dictionary.idx2item)]
+    sentence_list = cas.select(SENTENCE_TYPE)
+    list_sentences = []
+    for sentence in sentence_list:
+        tokens = []
+        pos_list = list(cas.select_covered(POS_TYPE, sentence))
+        token_list = list(cas.select_covered(TOKEN_TYPE, sentence))
+        if len(pos_list) != len(token_list):
+            continue
+        flag = False
```

Comment:
What does "flag" indicate? There is no comment explaining it and the variable name could mean anything.

Reply:
When there is a wrong value of POS which is not in the tagset, we need ignore this sentence. The "flag" is used to control this. Maybe I should use a more specific name.